search-sentiments-map/routes/search.js

```diff
@@ -0,0 +1,278 @@
+  } catch (err) {
+    console.error('ERROR:', err);
+  }
+}
+
+/** 
+ * Updates daily search results (accumulates by day) in the Datastore.
+ */
+function updateSearchResults() {
+  retrieveGlobalTrends().then(async trends => {
+    for (let i = 0; i < trends.length; i++) { // When testing, make i < 1 trend to test one.
+      await updateSearchResultsForTopic(trends[i].trendTopic);
+      // TODO(ntarn): Remove console.log statements when finished debugging.
+      console.log('ntarn debug updateSearchResult for: ' + trends[i].trendTopic);
+      await new Promise(resolve => setTimeout(resolve, 60000));
+    }
+  });
+}
+
+/** 
+ * Retrieves search result data for all countries for a given topic. Passes
+ * this data to be saved in Datastore.
+ * @param {string} query Search query.
+ */
+async function updateSearchResultsForTopic(query) {
+  let countriesData = [];
+
+  // Note: Can't use forEach with await.
+  for (let i = 0; i < json.length; i++) { // When testing, make i < 3 countries. 
+    var set = 0;
+    // 100 queries per minute limit for Custom Search API. Pause to prevent
+    // surpassing limit.
+    if (i !== 0 && i % 100 === 0) {
+      await new Promise(resolve => setTimeout(resolve, 60000));
+    }
+    let countryData = [];
+    // Update countryData within the functions called.
+    const avg = await getSearchResultsForCountryFromAPI(
+      "country" + json[i].id, query, countryData);
+    console.log('ntarn debug country: ' + json[i].id + ' averageSentiment: ' + avg);
+    countriesData.push({
+      country: json[i].id,
+      averageSentiment: avg,
+      results: countryData,
+    });
+  }
+  addTopicToDatastore(query, countriesData);
+}
+
+/** 
+ * Gets top 10 search results for specified country and query from Custom
+ * Search API and passes them to be saved in countryData object.
+ * @param {string} countryCode 2 letter country code for search results to be
+ *     written in.
+ * @param {string} query Search query.
+ * @param {Object} countryData Object holding all searchResults for a country.
+ */
+async function getSearchResultsForCountryFromAPI(countryCode, query, countryData) {
+  let response =
+    await fetch('https://www.googleapis.com/customsearch/v1?key=AIzaSyDszWv1aGP7Q1uOt74CqBpx87KpkhDR6Io&cx=017187910465527070415:o5pur9drtw0&q=' + query + '&cr=' + countryCode + '&num=10&safe=active&dateRestrict=d1&fields=items(title,snippet,htmlTitle,link)');
+  let searchResults = await response.json();
+  return await saveResultsAndDeletePrevious(searchResults, countryData);
+}
+
+/**
+ * Deletes previous search results and saves the current results in the
+ * Datastore, given the results JSON obtained from the API.
+ * @param {Object} searchResultsJson Object with information for top 10 search
+ *     results.
+ * @param {Object} countryData Object holding all searchResults for a country.
+ */
+async function saveResultsAndDeletePrevious(searchResultsJson, countryData) {
+  await deleteAncientResults();
```

Comment:
Call `deleteAncientResults` in `updateSearchResults` because you don't want to call this for every trend. Rename this function to something like `formatResults` because you are not actually saving it to Datastore. Also, change the comment/

Reply:
@carmenBen